src/lib/schemas/deep-research.ts

```diff
@@ -10,11 +10,24 @@ export const DeepResearchJobInputSchema = z.object({
 
 export type DeepResearchJobInput = z.infer<typeof DeepResearchJobInputSchema>;
 
-// Since we can't define a circular type directly in Zod, we use a more generic structure
-// that can represent our RerankResult<SearchGroups> type
+// Since we can't define a circular type directly in Zod, we use a more specific structure
+// that represents our RerankResult item structure
+const ItemSchema = z.object({
+  id: z.string().optional(),
```

Comment:
This should not be optional everywhere. The least amount of optional types as possible, so we enforce as much of the schema as we can. Look at what's strictly required by the formatter (since that's where it's used in the end) and model the schema here after the minimum requirements there.